### ISSUE TITLE
Doesn't flatpak allow extension updates to switch branch when the extension point changes version?

### ISSUE THREAD

**Issue by Jehan** [flathub/org.gimp.GIMP#105]
@hfiguiere From our discussions on https://github.com/flathub/org.gimp.GIMP/pull/98#issuecomment-808351795, I got that:

* When we update the runtime, we should increment the plug-in extension point with the scheme you made: https://github.com/flathub/org.gimp.GIMP/blob/master/org.gimp.GIMP.json#L45
* Then all plug-in extensions will have to bump the "*sdk*" version to the same as our runtime and increment their "*branch*". For instance G'MIC: https://github.com/flathub/org.gimp.GIMP.Plugin.GMic/blob/master/org.gimp.GIMP.Plugin.GMic.json#L3

Then if I understood correctly, GIMP can only see the plug-in extensions in the set branch (i.e. only "2-40"). Right?

# The problem

Apparently several people are complaining that they "lost" their plug-ins, but actually they have to manually reinstall them on the new branch. So they have to specifically reinstall `org.gimp.GIMP.Plugin.GMic//2-40` for instance. That implies they have to understand the technicalities or the fact that we updated the runtime (the kind of details most people don't need to know).

Also this means the old `org.gimp.GIMP.Plugin.GMic//2-3.36` is still there even if it is technically unusable (because the old GIMP using this branch is not anymore available since it was on the same `stable` branch). So we would slowly fill up space with outdated plug-ins too.

# Solution

Is there currently a solution for these issues?

In other words: `flatpak` should recognize that the extension point version changed, hence check currently installed plug-ins, check if the same plug-in (same `id`) exist in the new branch and propose this new `plug-in//branch` package. Also it should propose to remove the old one as it became unreachable.

Did I understand all correctly? Is there a solution (i.e. we don't do it right, currently)? If not, shouldn't we make a report at flatpak and ask for this to work like proposed? Thanks. 🙂

**Comment by hfiguiere** [comment]
I filed this flatpak bug
https://github.com/flatpak/flatpak/issues/4208

**Comment by hfiguiere** [comment]
Also filed https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1216

**Comment by brunvonlope** [comment]
Our extension point is now GIMP major version now so this bug can not be reproduced anymore (at least not between minor and micro gimp updates).

When updating to a new major version, since the API is changed, the plug-ins will naturally not work too like in all other platforms. So, for consistency, closing as fixed.
